client/network/src/service.rs

```diff
@@ -201,7 +201,12 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
 			params.chain.clone(),
 			params.protocol_id.clone(),
 			&params.network_config,
-			vec![default_notif_handshake_message; params.network_config.extra_sets.len()],
+			iter::once(Vec::new())
+				.chain(
+					(0..params.network_config.extra_sets.len() - 1)
```

Comment:
`.chain(vec![default_notif_handshake_message; params.network_config.extra_sets.len()-1])` should also work